packages/sdk/clients/createNexusClient.ts

```diff
@@ -0,0 +1,187 @@
+> = Prettify<
+  Pick<
+    ClientConfig<transport, chain, account, rpcSchema>,
+    | "account"
+    | "cacheTime"
+    | "chain"
+    | "key"
+    | "name"
+    | "pollingInterval"
+    | "rpcSchema"
+  > & {
+    /** RPC URL. */
+    transport: transport
+    /** Bundler URL. */
+    bundlerTransport: transport
+    /** Client that points to an Execution RPC URL. */
+    client?: client | Client | undefined
+    /** Paymaster configuration. */
+    paymaster?:
+      | true
+      | {
+          /** Retrieves paymaster-related User Operation properties to be used for sending the User Operation. */
+          getPaymasterData?: PaymasterActions["getPaymasterData"] | undefined
+          /** Retrieves paymaster-related User Operation properties to be used for gas estimation. */
+          getPaymasterStubData?:
+            | PaymasterActions["getPaymasterStubData"]
+            | undefined
+        }
+      | undefined
+    /** Paymaster context to pass to `getPaymasterData` and `getPaymasterStubData` calls. */
+    paymasterContext?: unknown
+    /** User Operation configuration. */
+    userOperation?:
+      | {
+          /** Prepares fee properties for the User Operation request. */
+          estimateFeesPerGas?:
+            | ((parameters: {
+                account: account | SmartAccount
+                bundlerClient: Client
+                userOperation: UserOperationRequest
+              }) => Promise<EstimateFeesPerGasReturnType<"eip1559">>)
+            | undefined
+        }
+      | undefined
+    /** Owner of the account. */
+    holder: UnknownHolder
+    /** Index of the account. */
+    index?: bigint
+    /** Active module of the account. */
+    activeModule?: BaseValidationModule
+    /** Factory address of the account. */
+    executorModule?: BaseExecutionModule
+    factoryAddress?: Address
+    /** Owner module */
+    k1ValidatorAddress?: Address
+  }
+>
+
+export async function createNexusClient(
+  parameters: NexusClientConfig
+): Promise<NexusClient> {
+  const {
+    client: client_,
+    chain = parameters.chain ?? client_?.chain,
+    holder,
+    index = 0n,
+    key = "nexus client",
+    name = "Nexus Client",
+    activeModule,
+    executorModule,
+    factoryAddress = contracts.k1ValidatorFactory.address,
+    k1ValidatorAddress = contracts.k1Validator.address,
+    bundlerTransport,
+    paymaster,
+    transport,
+    paymasterContext,
+    userOperation = {
+      estimateFeesPerGas: async (parameters) => {
+        const feeData = await (
+          parameters?.account?.client as PublicClient
+        )?.estimateFeesPerGas?.()
+        return {
+          maxFeePerGas: feeData.maxFeePerGas * 2n,
+          maxPriorityFeePerGas: feeData.maxPriorityFeePerGas * 2n
+        }
+      }
+    }
+  } = parameters
+
+  if (!chain) throw new Error("Missing chain")
+
+  const nexusAccount = await toNexusAccount({
+    transport,
+    chain,
+    holder,
+    index,
+    activeModule,
+    executorModule,
```

Comment:
where does this play a role?

Reply:
do we still have notion of default validation module.
active module can also be renamed to activeValidator

Reply:
The executorModule doesn't do anything yet - that's the case both before and after this PR. Gabi is currently working on this

Reply:
ok. will review post merge then

Reply:
the only active module type to set on the SDK would be a validator, I removed the part where we set active executor as it doesn't make sense

Reply:
removed